### PR TITLE
Initial implementation of Resources template

### PR DESCRIPTION
**Pull request by comzeradd**

### CODE REVIEW

File: coss/opensource_clubs/models.py [hunk 1]
```diff
@@ -158,6 +158,45 @@ class FAQPage(Page):
...
+    ])
+    resources_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                       max_length=50)
+    resources_cta_text = models.CharField(verbose_name='CTA Text', blank=True, default='',
+                                          max_length=50)
+    resources_cta_link = models.URLField(verbose_name='Link', blank=True, default='')
+    guides = fields.RichTextField(verbose_name='Guides', blank=True)
+
+    def get_mentors(self):
+        return ClubProfile.objects.filter(is_mentor=True)
```
Comment (akatsoulas) [inline coss/opensource_clubs/models.py]: Let's add a todo here to filter the users that belong to each club.

File: coss/opensource_clubs/models.py [hunk 2]
```diff
@@ -175,3 +214,4 @@ class ClubProfile(models.Model):
     """User profile relevant only to the Open Source Clubs."""
     user = models.OneToOneField(UserModel)
     is_captain = models.BooleanField(default=False)
+    is_mentor = models.BooleanField(default=False)
```
Comment (akatsoulas) [inline coss/opensource_clubs/models.py]: Do we know if mentors will only be members of a club or is there a chance that we will have external mentors (eg staff)?

File: coss/opensource_clubs/templates/opensource_clubs/resources_page.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  <img src="{% static 'img/clubs-white.svg' %}" alt="mozilla">
+{% endblock %}
+
+{% block jumbotron %}
+{% endblock %}
+
+{% block content %}
+  <div class="content">
+    {% if page.heading_text %}
+      {% if page.heading_image %}
```
Comment (akatsoulas) [inline coss/opensource_clubs/templates/opensource_clubs/resources_page.html]: Let's move the two `if`s in the same line with an `and` operator.

File: coss/opensource_clubs/models.py (hunk 2 above)
Comment (comzeradd) [reply coss/opensource_clubs/models.py]: The latter. Based on the Content Outline mentors are not necessarily members of any Club. I added this here, so ideally they have a profile on the website.

File: coss/opensource_clubs/models.py (hunk 1 above)
Comment (comzeradd) [reply coss/opensource_clubs/models.py]: Do we need that in this specific method? Resources page content scope is for all clubs and mentors listed here are for all Club members. See also below.

File: coss/opensource_clubs/templates/opensource_clubs/resources_page.html (hunk 3 above)
Comment (comzeradd) [reply coss/opensource_clubs/templates/opensource_clubs/resources_page.html]: By doing that we'll miss rendering that block in cases where we have text but not an image.

File: coss/opensource_clubs/models.py (hunk 2 above)
Comment (akatsoulas) [reply coss/opensource_clubs/models.py]: This feels a bit problematic, to ask all the mentors to have a profile that is. In any case, let's leave this as is for now.

File: coss/opensource_clubs/models.py [hunk 4]
```diff
@@ -158,6 +158,45 @@ class FAQPage(Page):
...
+    heading_text = fields.RichTextField(verbose_name='Text', blank=True)
+    heading_image = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+',
+        verbose_name='Image'
+    )
+    mentors = fields.StreamField([
```
[the remark this answers is not in the document]
Comment (comzeradd) [reply coss/opensource_clubs/models.py]: I don't think it's necessary. It's a StreamField already, so you can add multiple `InputTextContentBlock` blocks.

[the remark this answers is not in the document]
Comment (akatsoulas) [reply coss/opensource_clubs/models.py]: InputTextContentBlock is basically a few fields to input text. How can we associate mentor with profiles if we use this?

[the remark this answers is not in the document]
Comment (comzeradd) [reply coss/opensource_clubs/models.py]: Maybe I should rename the field to make this more clear. This field doesn't need to be associated with mentor profiles. It's just the content block on top of them. See the [design](https://invis.io/AZDAP6MPT#/251381672_General_Content_Template_-_Resources). It's the introductory section above mentors profiles.

[the remark this answers is not in the document]
Comment (akatsoulas) [reply coss/opensource_clubs/models.py]: Ah, good point. Let's rename this to something more relevant (eg mentors_description)

File: coss/opensource_clubs/models.py (hunk 1 above)
Comment (akatsoulas) [reply coss/opensource_clubs/models.py]: I believe we do. I suspect that different clubs will have different mentors. That said, we cannot do much at the moment so let's add a `TODO` as a placeholder.

**Pull request merged**